### PR TITLE
Clarify that we want isSameDay(…, 'London') in the DateRange component

### PR DESCRIPTION
**Pull request by alexwlchan**
For https://github.com/wellcomecollection/wellcomecollection.org/issues/7831

We use `isSameDay()` in the DateRange component, and I wasn't sure if we want London or UTC. I wrote some additional documentation and tests, and that made it clear to me that we want a London-based comparison (although admittedly it's an edge case).

* We only render the times if the start/end are on the same day
* The times are rendered in a London timezone

There are two tests for the edge cases where this would cause weirdness if we don't do a London isSameDay() comparison:

* same London day, different UTC day
* different London day, same UTC day

The component itself is only a one-line change; the rest of the patch is the context that convinced me it was the right change to make.

### CODE REVIEW

File: common/views/components/DateRange/DateRange.test.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { mountWithTheme } from '@weco/common/test/fixtures/enzyme-helpers';
+import DateRange from './DateRange';
+
+describe('DateRange', () => {
+  it('renders a date + time range if the start/end are on the same day', () => {
+    const start = new Date('2022-09-18T12:00:00+0100');
+    const end   = new Date('2022-09-18T14:30:00+0100'); // prettier-ignore
```
Comment (alexwlchan) [inline common/views/components/DateRange/DateRange.test.tsx:7]: note: the `// prettier-ignore` comment means prettier doesn't remove the whitespace between `end =`, so the timestamps line up nicely.

File: common/views/components/DateRange/DateRange.test.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    const component = mountWithTheme(<DateRange start={start} end={end} />);
+    expect(component.html()).toEqual(
+      '<time datetime="2022-09-18T11:00:00.000Z">Sunday 18 September 2022</time>, ' +
+        '<span><time datetime="2022-09-18T11:00:00.000Z">12:00</time> – <time datetime="2022-09-18T13:30:00.000Z">14:30</time></span>'
```
Comment (alexwlchan) [inline common/views/components/DateRange/DateRange.test.tsx:12]: note: I did consider removing the extraneous `<span>`, but that felt like an unnecessary change.

File: common/views/components/DateRange/DateRange.test.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    //                             -18T23:30:00 UTC
+    const end   = new Date('2022-09-19T00:30:00+0100'); // prettier-ignore
+
+    const component = mountWithTheme(<DateRange start={start} end={end} />);
+    expect(component.html()).toEqual(
+      '<time datetime="2022-09-18T22:30:00.000Z">18 September 2022</time> – ' +
+        '<time datetime="2022-09-18T23:30:00.000Z">19 September 2022</time>'
+    );
+  });
```
Comment (alexwlchan) [inline common/views/components/DateRange/DateRange.test.tsx:74]: These two tests fail if you use a UTC comparison. In particular you get:

```
17 September 2022 – 18 September 2022
```

and

```
18 September 2022, 23:30 – 00:30
```

both of which are wrong.

File: common/views/components/DateRange/DateRange.test.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    );
+    expect(component.html()).toEqual(
+      '<time datetime="2022-09-18T11:00:00.000Z">Sunday 18 September 2022</time>' +
+        '<span class="block"><time datetime="2022-09-18T11:00:00.000Z">12:00</time> – <time datetime="2022-09-18T13:30:00.000Z">14:30</time></span>'
+    );
+  });
+
+  it('renders a date + time range if the start/end are on the same London day', () => {
+    // It seems unlikely we would ever schedule an event this early in the
+    // day, but let’s make sure we handle it properly.  Note that these two
```
Comment (davidpmccormick) [inline common/views/components/DateRange/DateRange.test.tsx:42]: I didn't have you down as a two-spaces-after-a-full-stop kind of person!

**Review by davidpmccormick — APPROVED** [review]

File: common/views/components/DateRange/DateRange.test.tsx (hunk 4 above)
Comment (alexwlchan) [reply common/views/components/DateRange/DateRange.test.tsx:42]: spaaaaaaaace
